### PR TITLE
Warn users of increased `recombine_submeshes` memory demand.

### LINKED ISSUE #4845: Performance Shift(s): `bbe75acd`
Raising on behalf of the GHA workflow, which was having troubles at the time this originally came up!

---

Benchmark comparison has identified performance shifts at

* commit bbe75acd (#4831).

Please review the report below and take corrective/congratulatory action as appropriate :slightly_smiling_face:

<details>
<summary>Performance shift report</summary>

```
       before           after         ratio
     [0efd78d9]       [bbe75acd]

+       20.203125      54.70703125     2.71  experimental.ugrid.regions_combine.CombineRegionsComputeRealData.track_addedmem_compute_data(500)
```

</details>

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #4845 (read that for details).

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -145,8 +145,8 @@ This document explains the changes made to Iris for this release
    code that explicitly checks the calendar attribute. (:pull:`4847`)
 
 
-🚀 Performance Enhancements
-===========================
+🚀 Performance
```
Comment (pp-mo) [inline docs/src/whatsnew/latest.rst:148]: quiet change of scope here :grin: :+1:

File: docs/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -166,6 +166,10 @@ This document explains the changes made to Iris for this release
 #. `@wjbenfold`_ improved the speed of linear interpolation using
    :meth:`iris.analysis.trajectory.interpolate` (:pull:`4366`)
 
+#. NumPy ``v1.23`` behaviour changes mean that
+   :func:`iris.experimental.ugrid.utils.recombine_submeshes` now uses ~3x as
+   much memory - up to 700MB for the largest expected data chunks. Investigated
```
Comment (pp-mo) [inline docs/src/whatsnew/latest.rst:171]: I don't think we can go into the details of how you might calculate this.
But also we can't really justify  "largest expected data" for general usage.
So, can we just indicate the rough size of a testcase that takes X memory?

**Review by pp-mo — CHANGES_REQUESTED** [review]

File: docs/src/whatsnew/latest.rst (hunk 2 above)
Comment (trexfeathers) [reply docs/src/whatsnew/latest.rst]: 4ffc94ddccf3dea8375bc51a73584bfe1d7bd646

Comment (pp-mo) [reply docs/src/whatsnew/latest.rst]: :+1: sorted !

**Review by pp-mo — APPROVED** [review]

**Pull request merged**